File: src/com/l2kt/gameserver/scripting/tasks/ServerShutdown.java
package com.l2kt.gameserver.scripting.tasks;

import com.l2kt.gameserver.Shutdown;
import com.l2kt.gameserver.scripting.ScheduledQuest;

public final class ServerShutdown extends ScheduledQuest
{
	private static final int PERIOD = 600; // 10 minutes
	
	public ServerShutdown()
	{
		super(-1, "tasks");
	}
	
	@Override
	public final void onStart()
	{
		new Shutdown(PERIOD, false).start();
	}
	
	@Override
	public final void onEnd()
	{
	}
}

File: src/com/l2kt/gameserver/scripting/tasks/ServerShutdown.kt
package com.l2kt.gameserver.scripting.tasks

import com.l2kt.gameserver.Shutdown
import com.l2kt.gameserver.scripting.ScheduledQuest

class ServerShutdown : ScheduledQuest(-1, "tasks") {

    public override fun onStart() {
        Shutdown(PERIOD, false).start()
    }

    public override fun onEnd() {}

    companion object {
        private const val PERIOD = 600 // 10 minutes
    }
}